mixin/thanos-mixin/README.md

```diff
@@ -0,0 +1,124 @@
+
+[embedmd]:# (config.libsonnet)
+```libsonnet
+{
+  _config+:: {
+    thanosQuerierJobPrefix: 'thanos-querier',
+    thanosStoreJobPrefix: 'thanos-store',
+    thanosReceiveJobPrefix: 'thanos-receive',
+    thanosRuleJobPrefix: 'thanos-rule',
+    thanosCompactJobPrefix: 'thanos-compact',
+    thanosSidecarJobPrefix: 'thanos-sidecar',
+
+    thanosQuerierSelector: 'job=~"%s.*"' % self.thanosQuerierJobPrefix,
+    thanosStoreSelector: 'job=~"%s.*"' % self.thanosStoreJobPrefix,
+    thanosReceiveSelector: 'job=~"%s.*"' % self.thanosReceiveJobPrefix,
+    thanosRuleSelector: 'job=~"%s.*"' % self.thanosRuleJobPrefix,
+    thanosCompactSelector: 'job=~"%s.*"' % self.thanosCompactJobPrefix,
+    thanosSidecarSelector: 'job=~"%s.*"' % self.thanosSidecarJobPrefix,
+
+    // We build alerts for the presence of all these jobs.
+    jobs: {
+      ThanosQuerier: $._config.thanosQuerierSelector,
+      ThanosStore: $._config.thanosStoreSelector,
+      ThanosReceive: $._config.thanosReceiveSelector,
+      ThanosRule: $._config.thanosRuleSelector,
+      ThanosCompact: $._config.thanosCompactSelector,
+      ThanosSidecar: $._config.thanosSidecarSelector,
+    },
+
+    // Config for the Grafana dashboards in the thanos-mixin
+    grafanaThanos: {
+      dashboardNamePrefix: 'Thanos / ',
+      dashboardTags: ['thanos-mixin'],
+
+      dashboardOverviewTitle: '%(dashboardNamePrefix)sOverview' % $._config.grafanaThanos,
+      dashboardCompactTitle: '%(dashboardNamePrefix)sCompact' % $._config.grafanaThanos,
+      dashboardQuerierTitle: '%(dashboardNamePrefix)sQuerier' % $._config.grafanaThanos,
+      dashboardReceiveTitle: '%(dashboardNamePrefix)sReceive' % $._config.grafanaThanos,
+      dashboardRuleTitle: '%(dashboardNamePrefix)sRule' % $._config.grafanaThanos,
+      dashboardSidecarTitle: '%(dashboardNamePrefix)sSidecar' % $._config.grafanaThanos,
+      dashboardStoreTitle: '%(dashboardNamePrefix)sStore' % $._config.grafanaThanos,
+    },
+  },
+}
+```
+
+You can format your code using:
+```shell
+$ make jsonnet-format
+```
+
+## Generate
+
+To generate examples after modifying, make sure `jsonnet` dependencies are installed.
+```shell
+$ make jsonnet-vendor
+```
+
+and then
+
+```shell
+$ make examples
+```
+
+Make action runs the jsonnet code, then reads each key of the generated json and uses that as the file name, and writes the value of that key to that file, and converts each json manifest to yaml.
+
+> Make commands should handle dependecies for you.
+
+> In case it doesn't; before compiling, make sure `gojsontoyaml` is installed tool with `go get github.com/brancz/gojsontoyaml`.
```

Comment:
Why are these useful things written as comments? :p

Reply:
I'm going to re-organize docs a bit.